src/parse_ncbi_xml.py

```diff
@@ -1,54 +1,93 @@
-from .util import *
-from Bio.Blast import NCBIXML
 import re
 import sys
-    
+
+from Bio.Blast import NCBIXML
+
+from .util import *
+
+HEADER = ("id,start,end,alignment_id,enzyme_id,enzyme_code,product,"
+"organism,existence,gene,score,evalue,identity_percentage,query_coverage\n")
+
+# csv column title
+HEADER_ELEMENT = ["id", "start", "end", "alignment_id", "enzyme_id",
+    "enzyme_code",  "product", "organism", "existence", "gene", "score",
+    "evalue", "identity_percentage", "query_coverage"]
+HEADER = ",".join(HEADER_ELEMENT)
+
+
+REGEX_ELEMENTS = re.compile("(.*) OS=(.*) OX=.* PE=(\\d)")
+# OS: organism
+# PE: protein existence
+
+REGEX_PRODUCT = re.compile("^(.+)~{3}(.+)~{3}(.+)$")
+REGEX_GENE = re.compile("GN=(.*) PE=")
+
+
+def parse_product_split(labels):
+    # if labeled: {enzyme_id}~~~{enzyme_code}~~~{product}
+    # if not    : {product}
+    labels_split = labels.replace(",", "").split("~~~")
+    try:
+        enzyme_id, enzyme_code, product_name = labels_split[0:4]
+    except ValueError:
+        enzyme_id, enzyme_code, product_name = None, None, labels_split[0]
+        # Assuming only 3 parts or others
+    return enzyme_id, enzyme_code, product_name
+
+
+def parse_product_regex(labels):
+    match = REGEX_PRODUCT.search(labels)
+    try:
+        enzyme_id, enzyme_code, product_name = match.groups()
+    except AttributeError:
+        if not "~~~" in labels:
+            enzyme_id, enzyme_code, product_name = None, None, labels
+        else:
+            raise ValueError("Invalid database product format")
+    return enzyme_id, enzyme_code, product_name
+
+def parse_gene(desc):
+    # TODO:
+    pass
+
+
+def parse_alignment_title(alignment_title):
+    alignment_id_full, description = alignment_title.split(" ", 2)[1:3]
+
+    alignment_id = alignment_id_full.split("|")[1]
+    element = REGEX_ELEMENTS.search(description)
+    labels, organism, existence = element.groups()
+
+    enzyme_id, enzyme_code, product_name = parse_product_split(labels)
+    enzyme_id, enzyme_code, product_name = parse_product_regex(labels)
```

Comment:
Are there any reasons that we need two separate functions to parse the product?
`parse_product_regex` itself seems enough to me.

Reply:
No, there are exactly the same just two different approaches.

Reply:
Maybe I'll pick the second one (`parse_product_regex`). 
There're only three cases that will happen: 1) can be split into 3 variables, 2) cannot be split, so only 1, 3) split into 2 or more than 3 variables. (so value errors should be raised)
The second function deals with these better.